### ISSUE TITLE
When `@see` directs to a function that is a keyword, Dokka does not display a "See also" link to the function

### ISSUE THREAD

**Issue by aSemy** [Kotlin/dokka#2880]
**Describe the bug**

I want to create a `@see` link to a Java function, [`Named.as()`](https://kafka.apache.org/33/javadoc/org/apache/kafka/streams/kstream/Named.html#as(java.lang.String)). `as` is a Kotlin keyword.

Here is a copy-paste example

```kotlin
/** @see [Temp.as] */
fun test() {}

class Temp {
  // 
  fun `as`() {}
}
```

When I define a link, it looks okay in IntelliJ

<img width="248" alt="image" src="https://user-images.githubusercontent.com/897017/219856836-4ffa7da0-f72f-46ee-afde-c658fa3745e5.png">

But the 'See also' entry in Dokka only links to the class, not the function.

<img width="409" alt="image" src="https://user-images.githubusercontent.com/897017/219856649-14294b58-0991-4798-8f60-f4c298c7a7ab.png">

If I wrap the `as` in backticks, 

```kotlin
/** @see [Temp.`as`] */
fun test() {}
```

then Dokka generates a 'See also' link, but it's funny looking.

1. it's not left-aligned
2. `as` is highlighted in orange

But at least the link works!

<img width="453" alt="image" src="https://user-images.githubusercontent.com/897017/219856673-f019f348-c6d5-4d62-a37b-13d302678087.png">

**Expected behaviour**

1. I create `@see` links to functions that use Kotlin keywords
2. 'See also' entries are aligned

**Installation**

- Dokka version: 1.7.20

**Comment by IgnatBeresnev** [comment]
> When I define a link, it looks okay in IntelliJ

It looks OK, but if you try to navigate to the referenced element - it will lead to `Temp`, not to `as`.

Which made me think that Dokka's problem is actually the problem in our common dependency (we share some of the KDoc-related code)

Quick debugging confirms it:

```kotlin
/**
 * @see [Temp.`as`] description
 */
fun test() {}
```

![2023-02-21_15-37-40](https://user-images.githubusercontent.com/22685910/220374740-6689c168-58f5-40d7-a4ec-d1c944390d97.png)

`subjectName` should be `Temp.as`, and `content` should be `description`. For some reason, the underlying parser chooses to put `Temp.as` into `content`.

This also explains the formatting you're seeing. It's a table with two columns. Left column is for the key (name), the right one is for the description. Here there's no key/name, only the description, so the left column is empty

TLDR: should be fixed in the Kotlin IDE dependency

**Comment by IgnatBeresnev** [comment]
Same for 

```kotlin
/**
 * @see [Temp.as] description
 */
fun test() {}
```

Calling `resolveKDocLink` (which is an external function from our common dependency) for `Temp.as` returns just the class


<details>
<summary>screenshot</summary>

![2023-02-21_16-02-21](https://user-images.githubusercontent.com/22685910/220381038-26244b4b-98ac-45da-b373-0d0ffabdfbd4.png)

</details>

Whereas calling it for `Temp.foo` returns the function:

<details>
<summary>screenshot</summary>

![2023-02-21_16-02-12](https://user-images.githubusercontent.com/22685910/220381248-cdec88d8-755a-40cc-88c7-c092abebafaa.png)

</details>
